t/app/model/defecttype.t

```diff
@@ -59,6 +68,44 @@ subtest 'Problem->defect_types behaves correctly' => sub {
     is $problem->defect_types->first->name, $general_defect_type->name, 'Correct defect type is returned for Traffic lights category';
 };
 
+subtest 'by_categories returns all defect types grouped by category' => sub {
+    FixMyStreet::App->model('DB::BodyArea')->create( { body => $oxfordshire, area_id => $area_id } );
```

Comment:
I don't understand what this line is for. `create_body_ok` already puts an entry in BodyArea – if your aim is for body ID and area ID not to match, remove the `id => 2237` from the `create_body_ok` call above, no need for it to be in two areas here.